plugins/media-keys/gsd-media-keys-manager.c

```diff
@@ -3144,6 +3152,40 @@ power_keyboard_ready_cb (GObject             *source_object,
         }
 }
 
+static gboolean
+lid_closed_set (GsdMediaKeysManager *manager)
+{
+        g_debug ("lid is now %s", manager->priv->lid_is_closed_tmp ? "closed" : "open");
+        manager->priv->lid_is_closed = manager->priv->lid_is_closed_tmp;
```

Comment:
I really don't like the `tmp` additional private variable. In reality you only get here in a timeout when the variable is `FALSE`. Since when the variable becomes true you also need to remove the source ID (see comment below), I would just do that in that code path instead of calling into this function, and only call here to "confirm" the lid is now open.